pkgs/development/r-modules/default.nix

```diff
@@ -1321,6 +1321,16 @@ let
       RGL_USE_NULL = "true";
     });
 
+    reticulate = old.reticulate.overrideAttrs (_: {
+      preConfigure = ''
+        if [ ! -z ''${python+x} ] ; then
+          substituteInPlace R/zzz.R --replace \
+            ".onLoad <- function(libname, pkgname) {" \
+            ".onLoad <- function(libname, pkgname) { Sys.setenv(RETICULATE_PYTHON='$python/bin/python');"
```

Comment:
can we change this to:

```
Sys.setenv(RETICULATE_PYTHON=Sys.getenv("RETICULATE_PYTHON", unset='$python/bin/python'))
```

the current way the user can override the env var in the R session only after loading reticulate and only from within the R session. The alternative way they can specify before loading reticulate and even from outside the R session.

Reply:
Was the goal not to force the pure declaration of the environment? Doing this defeats that purpose.

Reply:
to me the goal should be to ensure users don't get the wrong python by accident, not prevent them from using an alternative python. In the current version, if you start a pure shell, and then run:

```
export RETICULATE_PYTHON=<my-coworkers-python>
R
library(reticulate)
py_config()
```

you don't get your coworker's python, which is surprising. And not sufficient to stop you anyway, you can simply run:

```
R
library(reticulate)
Sys.setenv("RETICULATE_PYTHON" = "<my-coworkers-python>")
py_config()
```

I think having the former return the requested python is preferable, but ultimately either is fine. Whichever you think is better

Reply:
I too wish to have least surprise and maximum functionality of the original package. Above, we discussed that state in project directories like venvs should be overridden, but now you're suggesting state in environment variables should not be overridden. At this point I think it's unclear what the priority of different state is and if this is universally shared amongst reticulate users.

I note that in your example you don't get your coworker's python only if a python has been declared in the nix config, and as you note if it has been declared you can still force override it explicitly, albeit in a non-intuitive way.

Reply:
I think the line suggested by cfhammill achieves this. I don't know if it works in practice. But the logic is to set `RETICULATE_PYTHON` *only if it's not set* already. That's the least surprising, right?

Reply:
Yeah, my handwavy argument is that a pure shell/devenv protects you from envvars leaking in, but it doesn't protect you from ambient files in the cwd changing the software env. If an env variable is set we can assume it was intentionally set inside the shell.

Although typing this out, I realize I'm thinking in flakes, nix-shell is not pure by default. I don't mean to bikeshed, happy with whatever you and others think is best.

Reply:
It is also a bad assumption because of .Renviron and .Rprofile anyway